portal/static/js/coredata.js

```diff
@@ -1,54 +1,152 @@
+        };
+        this.getEthnicityData = function() {
+            var ethnicityFields = this.getEthnicityFields().filter(":checked");
+            if (!ethnicityFields.length) {
+                return false;
+            }
+            var ethnicityIDs = ethnicityFields.map(function() {
+                return {
+                    code: $(this).val(),
+                    system: SYSTEM_IDENTIFIER_ENUM.ethnicity_system
+                };
+            }).get();
+            return {
+                "url": SYSTEM_IDENTIFIER_ENUM.ethnicity,
+                "valueCodeableConcept": {"coding": ethnicityIDs}
+            };
+        };
+        this.getRaceData = function() {
+            var raceFields = this.getRaceFields().filter(":checked");
+            if (!raceFields.length) {
+                return false;
+            }
+            var raceIDs = raceFields.map(function() {
+                return {
+                    code: $(this).val(),
+                    system: SYSTEM_IDENTIFIER_ENUM.race_system
+                };
+            }).get();
+            return {
+                "url": SYSTEM_IDENTIFIER_ENUM.race,
+                "valueCodeableConcept": {"coding": raceIDs}
+            };
+        };
+        this.setDemoData = function() {
+            if (!this.demoFieldsEnabled()) {
+                return;
+            }
+            var raceData = this.getRaceData();
+            var ethnicityData = this.getEthnicityData();
+            var demoArray = {resourceType: "Patient", extension: []};
+            if (raceData) {
+                demoArray.extension.push(raceData);
+            }
+            if (ethnicityData) {
+                demoArray.extension.push(ethnicityData);
+            }
+            tnthAjax.putDemo(this.subjectId, demoArray);
+        };
+        this.getDemoData = function() {
+            if (!this.demoFieldsEnabled()) {
+                return;
+            }
+            tnthAjax.getDemo(this.subjectId);
+        };
+        this.initDemoFieldEvents = function() {
+            if (!this.demoFieldsEnabled()) {
+                return;
+            }
+            var self = this;
+            this.getRaceFields().on("click", function() {
+                self.setDemoData();
+            });
+            this.getEthnicityFields().on("click", function() {
+                self.setDemoData();
+            });
+        };
+        this.initContinueButtonEvent = function() {
+            // Class for both "done" and "skip" buttons
+            var self = this;
+            $(".continue-btn").on("click", function(event){
+                event.preventDefault();
+                $(this).attr("disabled", true);
+                $(".loading-indicator").show();
+                try {
+                    window.location.replace(self.getReturnAddress());
+                } catch(e) {
+                    self.setError(i18next.t("Error occurred when redirecting to destination url"));
```

Comment:
This is a very technical, specific error message.

I don't think it should necessarily be shown to a user, but would still be helpful to a developer for debugging (maybe change to log message?).